### PR TITLE
Roll dart to c688d0c0c3ad3dece3a79ce0e115d787a94707ea.

### PR DESCRIPTION
**Pull request by aam**
Changes since last roll:
```
c688d0c0c3 [frontend] When serializing compiled expression proc, clone type params first.
346895eb35 Choose the right platform in the command-line fasta tool in strong mode
51d530475a [frontend] Honor embedSourceText option in incremental compiler.
06792cb91f Remove StackListener.popList
69cd7cb43f [infra] Add compare_results.dart and update_flakiness.dart.
c3c4e63ce4 Adjusted spec to use package syntax, thus eliminating bnf.sty
0315707dec Add 'set' case to type_variable_conflict_test
00f2054242 Add --repeat flag to test.py, so tests can be run multiple times
cf5cdea808 Add J/K versions CommonElements and ElementEnvironment
1681241062 Remove _ClassEnsurer
a5b71813ab Use json expectation files in dynamic tests
d560887ae5 Fix TypeError in Observatory
867c5271be Debugging dart in methods with dynamic type arguments
2e3f17fa2b Add support for the new mixin syntax to supertype constraint checking and resynthesis.
929b2fbc55 Adjust status file to account for CustomIsolates test passing now.
f0974aee20 Remove entry_points option from the test harness (should fix the precompiler bot failure).
296e4aa7e6 Split mixin_declaration_inference_valid_classes_test.dart into multiple tests.
2c427d33fc Train the dartanalyzer on less source.
41273507b2 Detect references to instance variables without declared type.
a0776096b9 Cleanup some of the build rules to not use Dart 1 mode.
3463f94ed9 Delete any existing version of a DevFS file before overwriting the file
7763ec3cd1 Surface analyzer toplevel inference limitations as DDC errors.
a386c07281 [vm/aot] Add extra call specialization pass
878017481f Roll dart_style to 1.2.0.
a890d95a26 [vm/compiler] Avoid !(x>y) -> x<=y on fp operands.
bc79674bc1 [vm/aot] Fix inferred type of implicitly initialized final fields
7df4900a94 Train the dartanalyzer tool on generating summaries.
80c1b88626 Add copyright comments in resolution tests.
bbe49524be Cache mapping from subtypes names to files where they are subtyped.
cf67e402bd Add test case reproducing issues #34498 and #34500.
81ce969060 Enable implicit constructors with optional parameters in mixin applications.
cb331064e0 Compare AnalysisOptions signatures in analyzer_cli.
eb059a76f3 Keep analyzer-specific status entries together in language_2_analyzer.status
62ce86e20c Add --test-list and --repeat options to test.py, for deflaking
9e86c1a395 Normalize strong.status
3831da8743 Include all problems in expectations
a1bd50168e Make input validation optional
6fe272314d Tweak parameters of ProcessedOptions
0a5d7b866d Update status files after de984e58.
de984e58cb Type checking for redirecting factories.
4134b95a3d [vm/tfa] Streamline handling of tear-offs in TFA
d44ffba254 [vm/tfa] Stop accepting non-empty entry points json files
30f0026022 [VM runtime] Fix type test: C is a subtype of C<FutureOr> (fixes #34482).
bbe83465ea Test that type of the ListLiteral with two different enums is inferred.
37b41bd317 Update status files after d4148159cc39d3f38d8295cf79ed736c6337d7ac
d9fcaa33b1 Implement StreamTransformer.fromBind
abb392fc0d add Pubspec.dependencyOverrides
c733ed7fd7 firefix status for LayoutTests/fast/dom/Range/getClientRects-character_t01
d4148159cc Initial valid inference checks
dcabf2cb0e Test for mixin declaration isObject and tweaks for error verifier.
```

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -115,7 +115,7 @@ allowed_hosts = [
 ]
 
 deps = {
-  'src': 'https://github.com/flutter/buildroot.git' + '@' + '7141218c62890635ce966e2530b465ea4c55c3e2',
+  'src': 'https://github.com/flutter/buildroot.git' + '@' + 'e191609670284e0d62ef2309d073b5586ac7d4a9',
```
Comment (zanderso) [inline DEPS]: Is it possible to roll the buildroot and Dart in separate PRs?

Comment (aam) [reply DEPS]: Sure, what is the reason for that though I'm curious?

Comment (aam) [reply DEPS]: https://github.com/flutter/engine/pull/6286

**Comment by goderbauer** [comment]
FYI: This dart roll increased the Dart snapshot size for the Hello World app by 2KB, which is about 1%.

**Comment by zanderso** [comment]
@mraleph

**Pull request merged**